src/main/asciidoc/security-refresh/security.adoc

```diff
@@ -0,0 +1,773 @@
+
+Identity stores:
+
+1. The Database Identity Store
+2. The LDAP Identity Store
+
+== Custom authentication mechanisms and identity stores 
+
+When the provided authentication mechanisms and identity stores are not sufficient, we can easily define our own. The provided ones and our own ones use the exact same interfaces, and the system doesn't distinguish between calling our own one or one that was provided by the system.
+
+== Authentication mechanisms and identity stores from other APIs
+
+The Servlet specification defines the exact same Form and Basic authentication mechanisms. Authenticating with them will have the exact same result as authenticating with a Jakarta Security authentication mechanism. E.g. role checks will work exactly the same independent on which API was used to authenticate. 
+
+A Servlet authentication mechanism however will not necessarily consult a Jakarta Security identity store. This is server dependent. The identity store that is called is server dependent as well. Calling this server dependent identity store is possible from Jakarta Security, but as an advanced feature.
+
+Likewise, programmatic role checks can be done from various APIs, including Jakarta Security, Jakarta REST and Jakarta Servlet. These all return the same outcome, independent from whether authentication took place with a Jakarta Security Authentication Mechanism or a Servlet Authentication Mechanism.
+
+
+== Securing an endpoint with Basic Authentication
+
+In the following example we'll be securing a REST endpoint using Basic Authentication.
+
+You'll learn the following things;
+
+1. How to define security constraints
+2. How to set a provided authentication mechanisms 
+3. How to define (and implicitly set) a custom identity store
+4. How to use the Jakarta Security SecurityContext
+
+=== Write the application
+
+Let's start with defining a simple REST resource class for a `/rest/resource` endpoint:
+
+[source,java]
+----
+@Path("/resource")
+@RequestScoped
+public class Resource {
+
+    @Inject
+    private SecurityContext securityContext;
+
+    @GET
+    @Produces(TEXT_PLAIN)
+    public String getCallerAndRole() {
+        return
+            securityContext.getCallerPrincipal().getName() + " : " +
+            securityContext.isCallerInRole("user");
+    }
+
+}
+----
+
+This resource uses the injected Jakarta EE SecurityContext to obtain access to the current authenticated caller, which is represented by a `Principal` instance. For simplicity sake the code here does not check whether the caller principal is `null`. If the resource is also available to non-authenticated callers, this is something we have to do. The `getCallerPrincipal()` call would return `null` then.
+
+Also note that there is a Jakarta REST specific type that is also named `SecurityContext` and has similar methods as the ones we used here. From the point of view of Jakarta EE that is a deprecated type and replaced by the Jakarta Security version.
```

Comment:
There have been talks about deprecating it, but it's complicated. KARG argues that REST users have no problems, so it should not change. And other people argue that since REST is in core profile (and therefor in MicroProfile) and Jakarta Security is not, it can't be deprecated. It's a big mess in EE :(

Maybe we can eventually have wording like "in a web profile or full profile product this type should not be used" or something along those lines, but it's not there now.